snapteld.go

```diff
@@ -323,6 +334,25 @@ func action(ctx *cli.Context) error {
 	// Set Max Processors for snapteld.
 	setMaxProcs(cfg.GoMaxProcs)
 
+	// Write PID file, if configured.
+	log.Info("Config PID file", cfg.PIDFile)
+	if cfg.PIDFile != "" {
+		log.Info("Creating PID file", cfg.PIDFile)
+		f, err := os.OpenFile(cfg.PIDFile, os.O_CREATE|os.O_WRONLY, 0644)
+		if err != nil {
+			log.Error("Unable to create pidfile", err)
+		} else {
+			fmt.Fprintf(f, "%d\n", os.Getpid())
+			f.Close()
+			defer func() {
+				err := os.Remove(cfg.PIDFile)
```

Comment:
Sorry, I have a stupid question to ask.
Why delete the PIDFile?

Reply:
Most servers are expected to delete their PID file when the server shuts down — for example, [Apache removes its PID file when shutting down](https://httpd.apache.org/docs/trunk/stopping.html#gracefulstop).

However, since snapteld calls `os.Exit()` when it shuts down, I recently realized that this deferred function will not be executed, since `os.Exit` shuts down immediately without calling deferred functions.

Reply:
@cce 
Thanks for your explanation!
Yes, I think the the actual behavior is not your meaning.
Could you please update this PR according your design when you are free?